python/python/lance/arrow.py

```diff
@@ -189,18 +201,25 @@ def from_uris(cls, uris: Union[pa.StringArray, Iterable[Union[str, Path]]]):
         <lance.arrow.ImageURIArray object at 0x...>
         ['file::///tmp/1.png']
         """
-
-        if not isinstance(uris, pa.StringArray):
+        if isinstance(uris, (pa.StringArray, pa.LargeStringArray)):
+            pass
+        elif isinstance(uris, Iterable):
             uris = pa.array((str(uri) for uri in uris), type=pa.string())
         else:
             raise TypeError("Cannot build a ImageURIArray from {}".format(type(uris)))
 
-        return cls.from_storage(ImageURIType(), uris)
+        return cls.from_storage(ImageURIType(uris.type), uris)
 
-    def read_uris(self):
+    def read_uris(self, storage_type=pa.binary()) -> "EncodedImageArray":
         """
         Read the images from the URIs into memory and return an EncodedImageArray
 
+        Parameters
+        ----------
+        storage_type : pa.DataType, optional
+            The storage type to use for the encoded images. Default is pa.binary().
+            To support larger arrays, use pa.large_binary().
```

Comment:
Should we mention the exact limit here?  Otherwise people are going to be wondering "how large should it be before I need pa.large_binary"?

Maybe something like, "if you expect to process batches greater than 2GiB, then use pa.large_binary()"